osu.Game/Screens/OsuScreen.cs

```diff
@@ -114,9 +114,14 @@ private void createDependencies(IReadOnlyDependencyContainer dependencies)
             Mods = screenDependencies.Mods;
         }
 
-        protected BackgroundScreen Background => backgroundStack?.CurrentScreen as BackgroundScreen;
+        public void ApplyToBackground(Action<BackgroundScreen> action) => background.ApplyToBackground(action);
 
-        private BackgroundScreen localBackground;
+        /// <summary>
+        /// The background created and owned by this screen. May be null if the background didn't change.
+        /// </summary>
+        private BackgroundScreen ownedBackground;
+
+        private BackgroundScreen background;
```

Comment:
I'd annotate both of these as `[CanBeNull]`. Annotating the latter as such actually exposes a potential NRE in `ApplyToBackground()`.

Reply:
I guess. Although not sure what to do in the case of that NRE. It should fail. Would you want a custom exception (`InvalidOperationException("Attempted to apply to background before screen is pushed")`)?